protocol/query/method/greet_server.json

```diff
@@ -0,0 +1,46 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "https://raw.githubusercontent.com/dedis/popstellar/master/protocol/query/method/greet_server.json",
+  "description": "Sent by the server to the other server when it connects to it. It informs the other server about its public key and client / server endpoints",
+  "type": "object",
+  "additionalProperties": false,
+  "properties": {
+    "method": {
+      "description": "[String] operation to be performed by the query",
+      "const": "greet_server"
+    },
+
+    "params": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "public_key": {
+          "description": "[String] public key of the server",
+          "type": "string",
+          "contentEncoding": "base64"
+        },
+        "server_address": {
+          "description": "[String] canonical address of the server with a protocol prefix and the port number for connecting as a server",
+          "type": "string",
+          "pattern": "^.*:\\/\\/.*:\\d{0,5}\\/.*$"
+        },
+        "client_address": {
+          "description": "[String] canonical address of the server with a protocol prefix and the port number for connecting as a client",
+          "type": "string",
+          "pattern": "^.*:\\/\\/.*:\\d{0,5}\\/.*$"

```

Comment:
are you sure you don't want to enforce `ws` and `wss` URI schemes only ?

Reply:
Yes, thank you I added that.